### PR TITLE
[XHarness] Add the Mono.Data.Tds tests.

### PR DESCRIPTION
**Pull request by mandel-macaque**
This branch just allows to run the tests in the dll from mono. Tests
will fail until they are fixed in the mono source.

### CODE REVIEW

File: tests/Makefile [hunk 1]
```diff
@@ -12,7 +12,7 @@ MTOUCH=$(IOS_DESTDIR)/$(MONOTOUCH_PREFIX)/bin/mtouch
 UNIT_SERVER_DIR=$(TOUCH_UNIT_PATH)/Touch.Server
 UNIT_SERVER=$(UNIT_SERVER_DIR)/bin/Debug/Touch.Server.exe
 TEST_SUITES=monotouch-test link\ sdk link\ all dont\ link framework-test mini
-BCL_TEST_SUITES=mscorlib System System.Core System.Data System.Net.Http System.Numerics System.Runtime.Serialization System.Transactions System.Web.Services System.Xml System.Xml.Linq Mono.Security System.ComponentModel.DataAnnotations System.Json System.ServiceModel.Web Mono.Data.Sqlite
+BCL_TEST_SUITES=mscorlib System System.Core System.Data System.Net.Http System.Numerics System.Runtime.Serialization System.Transactions System.Web.Services System.Xml System.Xml.Linq Mono.Security System.ComponentModel.DataAnnotations System.Json System.ServiceModel.Web Mono.Data.Sqlite Mono.Data.Tds
```
Comment (spouliot) [inline tests/Makefile]: maybe we should one assembly per line to facilitate reviews ? @rolfbjarne thoughts ?

File: tests/xharness/Harness.cs [hunk 2]
```diff
@@ -282,7 +282,7 @@ void AutoConfigureMac ()
 				});
 			}
 
-			var bcl_suites = new string[] { "mscorlib", "System", "System.Core", "System.Data", "System.Net.Http", "System.Numerics", "System.Runtime.Serialization", "System.Transactions", "System.Web.Services", "System.Xml", "System.Xml.Linq", "Mono.Security", "System.ComponentModel.DataAnnotations", "System.Json", "System.ServiceModel.Web", "Mono.Data.Sqlite" };
+			var bcl_suites = new string[] { "mscorlib", "System", "System.Core", "System.Data", "System.Net.Http", "System.Numerics", "System.Runtime.Serialization", "System.Transactions", "System.Web.Services", "System.Xml", "System.Xml.Linq", "Mono.Security", "System.ComponentModel.DataAnnotations", "System.Json", "System.ServiceModel.Web", "Mono.Data.Sqlite", "Mono.Data.Tds" };
```
Comment (spouliot) [inline tests/xharness/Harness.cs]: same (one per line)

File: tests/xharness/Harness.cs [hunk 3]
```diff
@@ -295,7 +295,7 @@ void AutoConfigureIOS ()
 			var library_projects = new string [] { "BundledResources", "EmbeddedResources", "bindings-test", "bindings-test2", "bindings-framework-test" };
 			var fsharp_test_suites = new string [] { "fsharp" };
 			var fsharp_library_projects = new string [] { "fsharplibrary" };
-			var bcl_suites = new string [] { "mscorlib", "System", "System.Core", "System.Data", "System.Net.Http", "System.Numerics", "System.Runtime.Serialization", "System.Transactions", "System.Web.Services", "System.Xml", "System.Xml.Linq", "Mono.Security", "System.ComponentModel.DataAnnotations", "System.Json", "System.ServiceModel.Web", "Mono.Data.Sqlite" };
+			var bcl_suites = new string [] { "mscorlib", "System", "System.Core", "System.Data", "System.Net.Http", "System.Numerics", "System.Runtime.Serialization", "System.Transactions", "System.Web.Services", "System.Xml", "System.Xml.Linq", "Mono.Security", "System.ComponentModel.DataAnnotations", "System.Json", "System.ServiceModel.Web", "Mono.Data.Sqlite", "Mono.Data.Tds" };
```
Comment (spouliot) [inline tests/xharness/Harness.cs]: same (one per line)

File: tests/Makefile (hunk 1 above)
Comment (rolfbjarne) [reply tests/Makefile]: Yes, one per line sounds good to me (for all cases).

**Comment by mandel-macaque** [comment]
Failures are expected. We need to fix those in the mono source of look into the test to see why the connection cannot be performed. @spouliot @rolfbjarne will fix the one per line comment, once that is done, should we land it and add a bug for the failing tests?

**Comment by spouliot** [comment]
@mandel-macaque PR to remain open until mono has the fixes and can be bumped (in the PR) to get a green build

**Comment by mandel-macaque** [comment]
A PR that fixes the failing tests on mono can be found here: https://github.com/mono/mono/pull/5501

**Comment by mandel-macaque** [comment]
Closing PR to create one against d15-5

**Pull request closed without merging**